boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -171,6 +174,10 @@ gdtptr:
     .word HEX(17)   /* Limit */
     .long gdt       /* Base Address */
 
+PUBLIC _MultibootInfoPtr
+_MultibootInfoPtr:
```

Comment:
The problem is that we have to be 100% sure the memory area where this pointer points to will not be erased when we relocate.

Reply:
MultibootInfoPtr is at 0x0025D400, and freeldr gets relocated at 0x00040000. We read the data from it at the initialization, so memory manager with heap **in theory** couldn't rewrite this area. The other question is the size of freeldr binary. :thinking:

Reply:
Implemented relocation-proof algorithm for multiboot info and memory map structures.